src/Checkbox/index.js

```diff
@@ -9,66 +9,124 @@ import { Icon } from './Icon'
 import { Label } from './Label'
 import { Input } from './Input'
 
-const Checkbox = ({
-    onChange,
-    name,
-    label,
-    className,
-    indeterminate,
-    required,
-    checked = false,
-    disabled,
-    valid,
-    warning,
-    error,
-}) => (
-    <label
-        className={cx('base', className, {
+class Checkbox extends Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            focus: props.focus,
```

Comment:
Do we need to copy the prop to state?

https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-rendering

Reply:
Yes, this is intentional. This way the input element can be focused without user interaction.
Contrary to `autofocus` this also works after the page has loaded.

I could add a `componentDidUpdate` to check whether the `focus` prop has changed and update the state accordingly, although I'm not certain we actually need this. As far as I know, none of our apps re-focus an input element, so I'd rather not do this until explicitly needed.